File: BulletPlugin/src/BulletFactoryEntry.cpp
#include "StdAfx.h"
#include "..\include\BulletFactoryEntry.h"
#include "BulletElementDefinition.h"

namespace BulletPlugin
{

BulletFactoryEntry::BulletFactoryEntry(SimObjectBase^ instance, BulletElementDefinition^ element)
:instance(instance), 
element(element)
{
}

void BulletFactoryEntry::createProduct(BulletScene^ scene)
{
    element->createProduct(instance, scene);
}

void BulletFactoryEntry::createStaticProduct(BulletScene^ scene)
{
    element->createStaticProduct(instance, scene);
}

}

File: BulletWrapperPlugin/Construction/BulletFactoryEntry.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Engine.ObjectManagement;

namespace BulletPlugin
{
    public class BulletFactoryEntry
    {
        SimObjectBase instance;
        BulletElementDefinition element;

        public BulletFactoryEntry(SimObjectBase instance, BulletElementDefinition element)
        {
            this.instance = instance;
            this.element = element;
        }

        public void createProduct(BulletScene scene)
        {
            element.createProduct(instance, scene);
        }

        public void createStaticProduct(BulletScene scene)
        {
            element.createStaticProduct(instance, scene);
        }
    }
}
